app/shared/modals/reservation-info/ReservationInfoContainer.js

```diff
@@ -0,0 +1,49 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import actionsUI from 'actions';
+import { fetchReservation } from 'api/actions';
+import reservationModalSelector from './reservationInfoSelector';
+import ReservationInfo from './ReservationInfo';
+
+
+export class UnconnectedReservationInfoContainer extends React.Component {
+  static propTypes = {
+    fetchReservation: PropTypes.func.isRequired,
+    onHide: PropTypes.func.isRequired,
+    reservation: PropTypes.object,
+    reservationId: PropTypes.number, // eslint-disable-line
+    resource: PropTypes.object,
+    show: PropTypes.bool.isRequired,
+    unit: PropTypes.object,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (!nextProps.reservation && nextProps.reservationId) {
+      this.props.fetchReservation(nextProps.reservationId);
+    }
+  }
+
+  render() {
+    return (this.props.reservation ?
```

Comment:
- [ ] It would be better to have a Loader inside the modal that does this check. The benefits:
  - Modal opens immediately (now there is a slight delay before the reservation has been fetched from the API).
  - Modal closes with an animation (now it disappears immediately when the reservation is removed from the state).